setup.py

```diff
@@ -31,12 +31,12 @@
 ###
 
 import os
+import subprocess
 import sys
 import time
 import warnings
-import datetime
-import tempfile
-import subprocess
+
+from setuptools import setup
```

Comment:
Catch `ImportError` and explain how to install setuptools if it isn't available. (Recommend installing it with the package-manager first, but also explain how to install pip and run `pip3 install --upgrade setuptools` in case setuptools isn't available in the package-manager).

See the deleted distutils comment for examples of how to present it.

Reply:
I think this is best left to pyproject.toml which specifies setuptools as the build dependency.  Otherwise this is just giving probably incorrect instructions -- are you installing globally?  are you in a virtualenv or venv? (virtualenv at least automatically installs setuptools so something else is then wrong)  are you even using pip?

Reply:
Existing users are used to running `setup.py`, which doesn't read `pyproject.toml` if setuptools isn't installed.

>  Otherwise this is just giving probably incorrect instructions

Then the script should say they must install setuptools somehow, and give some hints/links to do it.

Reply:
Please feel free to edit this pull request, as I think you have something specific in mind that is probably best not filtered through me as well.